pkg/skaffold/docker/image.go

```diff
@@ -339,7 +339,11 @@ func GetBuildArgs(a *latest.DockerArtifact) []string {
 		if v == nil {
 			args = append(args, k)
 		} else {
-			args = append(args, fmt.Sprintf("%s=%s", k, *v))
+			value, err := evaluateBuildArgsValue(*v)
```

Comment:
@balopat How should I handle the error here?

Reply:
return with an error and propagate it. We should fail fast if there is an environment issue. 
`return nil, err`

Reply:
Propagated the error.